cnd/src/http_api/routes/index/mod.rs

```diff
@@ -19,26 +19,31 @@ pub struct InfoResource {
     listen_addresses: Vec<Multiaddr>,
 }
 
-pub fn get_info<S: Network>(id: PeerId, dependencies: Facade<S>) -> Result<impl Reply, Rejection>
-where
-    S: Send + Sync + 'static,
-{
-    let listen_addresses: Vec<Multiaddr> = Network::listen_addresses(&dependencies).to_vec();
+pub fn get_info(
+    id: PeerId,
+    dependencies: Facade,
+) -> impl Future<Item = impl Reply, Error = Rejection> {
+    get_info_async(id, dependencies).boxed().compat()
+}
+
+async fn get_info_async(id: PeerId, facade: Facade) -> anyhow::Result<impl Reply, Rejection> {
+    let listen_addresses: Vec<Multiaddr> = facade.listen_addresses().await.to_vec();
 
     Ok(warp::reply::json(&InfoResource {
         id: Http(id),
         listen_addresses,
     }))
 }
 
-pub fn get_info_siren<S: Network>(
+pub fn get_info_siren(
```

Comment:
I am not sure to see the added value of going async but still having a wrapping compat layer here for such simple functions.

Reply:
These functions will eventually be async once we upgrade warp.